src/client/Microsoft.Identity.Client/Internal/Requests/Interactive/InteractiveRequest.cs

```diff
@@ -112,9 +113,9 @@ private async Task<MsalTokenResponse> GetTokenResponseAsync(CancellationToken ca
                 cancellationToken.ThrowIfCancellationRequested();
             }
 
-            if (_requestParams.AppConfig.MultiCloudSupportEnabled)
+            if (_requestParams.AppConfig.MultiCloudSupportEnabled && !_requestParams.AppConfig.ExtraQueryParameters.ContainsKey(_instanceAware))
```

Comment:
So the MultiCloudSupportEnabled is set internally by MSAL and the user needs to ensure "instance_aware" is added via WithExtraQueryParameters() api?

Reply:
No we add it to the extra query param to the interactive request. User just sets WithMultiCloudSupport(true)

Reply:
ok cool